vavr/src/test/java/com/baeldung/vavr/collections/CollectionAPIUnitTest.java

```diff
@@ -0,0 +1,261 @@
+                assertEquals(new Integer(4), k);
+                return v.dequeue();
+            })
+            .map((k, v) -> {
+                assertEquals(new Integer(5), k);
+                assertTrue(v.isEmpty());
+                return null;
+            });
+    }
+
+    @Test
+    public void givenStream_whenProcessed_thenCorrect() {
+        Stream<Integer> intStream = Stream.iterate(0, i -> i + 1)
+            .take(10);
+
+        assertEquals(10, intStream.size());
+
+        long evenSum = intStream.filter(i -> i % 2 == 0)
+            .sum()
+            .longValue();
+
+        assertEquals(20L, evenSum);
+        assertEquals(new Integer(5), intStream.get(5));
+    }
+
+    @Test
+    public void givenArray_whenQueried_thenCorrect() {
+        Array<Integer> intArray = Array.of(1, 2, 3);
+        Array<Integer> newArray = intArray.removeAt(1);
+
+        assertEquals(3, intArray.size());
+        assertEquals(2, newArray.size());
+
+        assertEquals(new Integer(1), intArray.get(0));
+        assertEquals(new Integer(2), intArray.get(1));
+        assertEquals(new Integer(3), intArray.get(2));
+        assertEquals(new Integer(3), newArray.get(1));
+    }
+
+    @Test
+    public void givenVector_whenQueried_thenCorrect() {
+        Vector<Integer> intVector = Vector.range(1, 5);
+        Vector<Integer> newVector = intVector.replace(2, 6);
+
+        assertEquals(4, intVector.size());
+        assertEquals(4, newVector.size());
+
+        assertEquals(new Integer(1), intVector.get(0));
+        assertEquals(new Integer(2), intVector.get(1));
+        assertEquals(new Integer(3), intVector.get(2));
+        assertEquals(new Integer(6), newVector.get(1));
+    }
+
+    @Test
+    public void givenCharSeq_whenProcessed_thenCorrect() {
+        CharSeq chars = CharSeq.of("vavr");
+        CharSeq newChars = chars.replace('v', 'V');
+
+        assertEquals(4, chars.size());
+        assertEquals(4, newChars.size());
+
+        assertEquals('v', chars.charAt(0));
+        assertEquals('V', newChars.charAt(0));
+        assertEquals("Vavr", newChars.mkString());
+    }
+
+    @Test
+    public void givenHashSet_whenModified_thenCorrect() {
+        HashSet<String> set = HashSet.of("Red", "Green", "Blue");
+        HashSet<String> newSet = set.add("Yellow");
+
+        assertEquals(3, set.size());
+        assertEquals(4, newSet.size());
+        assertFalse(set.contains("Yellow"));
+        assertTrue(newSet.contains("Yellow"));
+    }
+
+    @Test
+    public void givenSortedSet_whenIterated_thenCorrect() {
+        SortedSet<String> set = TreeSet.of("Red", "Green", "Blue");
+
+        Iterator<String> iterator = set.iterator();
+        assertEquals("Blue", iterator.next());
+        assertEquals("Green", iterator.next());
+        assertEquals("Red", iterator.next());
+    }
+
+    @Test
+    public void givenSortedSet_whenReversed_thenCorrect() {
+        Comparator<String> reverseCompare = (a, b) -> b.compareTo(a);
```

Comment:
the `(a, b) -> b.compareTo(a)` can be replaced with `Comparator.reverseOrder();`